app/uk/gov/hmrc/tai/util/yourTaxFreeAmount/IabdTaxCodeChangeReasons.scala

```diff
@@ -26,13 +26,33 @@ class IabdTaxCodeChangeReasons(totalTax: TotalTax)  {
 
   def reasons(iabdPairs: AllowancesAndDeductionPairs)(implicit messages: Messages): Seq[String] = {
 
-    val combinedBenefits = iabdPairs.allowances ++ iabdPairs.deductions
+  allowanceReasons(iabdPairs) ++ deductionReasons(iabdPairs)
+
+}
+
+private def allowanceReasons(iabdPairs: AllowancesAndDeductionPairs)(implicit messages: Messages): Seq[String] = {
+  val whatsChangedPairs = iabdPairs.allowances.filter(pair => pair.previous.isDefined && pair.current.isDefined)
+  val whatsNewPairs = iabdPairs.allowances.filter(pair => pair.previous.isEmpty && pair.current.isDefined)
+
+  whatsNewPairs.flatMap(translateNewBenefits(_)) ++
+    whatsChangedPairs.flatMap(translateChangedBenefits(_))
+}
+
+  private def deductionReasons(iabdPairs: AllowancesAndDeductionPairs)(implicit messages: Messages): Seq[String] = {
+  val whatsChangedPairs = iabdPairs.deductions.filter(pair => pair.previous.isDefined && pair.current.isDefined)
+  val whatsNewPairs = iabdPairs.deductions.filter(pair => pair.previous.isEmpty && pair.current.isDefined)
+  val whatsNewUnderpaymentPairs = whatsNewPairs.filter(pair => pair.componentType == UnderPaymentFromPreviousYear || pair.componentType == EstimatedTaxYouOweThisYear || pair.componentType == EarlyYearsAdjustment)
+  val whatsNewOtherPairs = whatsNewPairs.filterNot(pair => pair.componentType == UnderPaymentFromPreviousYear || pair.componentType == EstimatedTaxYouOweThisYear || pair.componentType == EarlyYearsAdjustment)
+
```

Comment:
Extract ` pair.componentType == UnderPaymentFromPreviousYear || pair.componentType == EstimatedTaxYouOweThisYear || pair.componentType == EarlyYearsAdjustment` into a function

could extract `pair.previous.isEmpty && pair.current.isDefined` and `pair.previous.isDefined && pair.current.isDefined` to a function as well if it helps wording what you are filtering for better?

Reply:
Done now. It definitely makes the filtering more readable.